pkg/operator/controller/ingress/controller.go

```diff
@@ -412,19 +412,70 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, infraConfig
 				changed = true
 			}
 
-			// If the ProviderParameters field does not exist for spec or status,
-			// just propagate (or remove) ProviderParameters in its entirety
-			// (as long as GCP parameters are specified one way or the other).
-			if specLB.ProviderParameters == nil && statusLB.ProviderParameters != nil && statusLB.ProviderParameters.GCP != nil ||
-				specLB.ProviderParameters != nil && specLB.ProviderParameters.GCP != nil && statusLB.ProviderParameters == nil {
-				ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters = specLB.ProviderParameters
-				changed = true
+			// Detect changes to provider-specific parameters.
+			// Currently the only platforms with configurable
+			// provider-specific parameters are AWS and GCP.
+			var lbType operatorv1.LoadBalancerProviderType
+			if specLB.ProviderParameters != nil {
+				lbType = specLB.ProviderParameters.Type
 			}
-
-			if specLB.ProviderParameters != nil && statusLB.ProviderParameters != nil &&
-				specLB.ProviderParameters.GCP != statusLB.ProviderParameters.GCP {
-				ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.GCP = specLB.ProviderParameters.GCP
-				changed = true
+			switch lbType {
+			case operatorv1.AWSLoadBalancerProvider:
+				// The only provider parameter that is supported
+				// for AWS is the connection idle timeout for
+				// classic ELBs.
+				var specIdleTimeout, statusIdleTimeout metav1.Duration
+				if specLB.ProviderParameters != nil && specLB.ProviderParameters.AWS != nil && specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
+					if v := specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout; v != nil {
+						specIdleTimeout = *v
+					}
+				}
+				if statusLB.ProviderParameters != nil && statusLB.ProviderParameters.AWS != nil && statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
+					if v := statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout; v != nil {
+						statusIdleTimeout = *v
+					}
+				}
+				if specIdleTimeout != statusIdleTimeout {
+					if statusLB.ProviderParameters == nil {
+						statusLB.ProviderParameters = &operatorv1.ProviderLoadBalancerParameters{
+							Type: operatorv1.AWSLoadBalancerProvider,
+						}
+					}
+					if statusLB.ProviderParameters.AWS == nil {
+						statusLB.ProviderParameters.AWS = &operatorv1.AWSLoadBalancerParameters{}
+					}
+					if statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters == nil {
+						statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters = &operatorv1.AWSClassicLoadBalancerParameters{}
+					}
+					var v *metav1.Duration
+					if specIdleTimeout.Duration > 0 {
```

Comment:
can `specIdleTimeout` be nil?

Reply:
No, it isn't a pointer type—it's declared as  `var specIdleTimeout, statusIdleTimeout metav1.Duration`.

(BTW, the line number that you are asking about is also the number of this PR!)